### PR TITLE
feat(backend): add asciidoc

### PR DESCRIPTION
**Pull request by cathaysia**

### CODE REVIEW

**Review by stevearc — CHANGES_REQUESTED** [review]
Can you add tests? Add an asciidoc file to [tests/](https://github.com/stevearc/aerial.nvim/tree/master/tests) and write a test that calls `util.test_file_symbols`. Here's where we're doing it for the markdown backend:

https://github.com/stevearc/aerial.nvim/blob/d21482d3be3228dde594aba31106db428292742c/tests/treesitter_spec.lua#L38-L42

It's embedded in the treesitter test because we use the same source file. You can put this in a standalone test file. Once you do that, run `./run_tests.sh --update` to generate a symbols snapshot file.

**Comment by cathaysia** [comment]
I add a test for asciidoc, but it failed in my computer. Could you give me some suggest?

**Review by stevearc — CHANGES_REQUESTED** [review]
I've fixed the test for you, but left a comment about code block parsing.

(note: the objdump test is still failing, but that's unrelated to this PR)

File: lua/aerial/backends/asciidoc.lua [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        then
+          table.insert(items, item)
+        end
+      end
+      while #stack > level and #stack > 0 do
+        table.remove(stack, #stack)
+      end
+      table.insert(stack, item)
+    elseif string.find(line, "```") == 1 then
+      inside_code_block = not inside_code_block
```
[the remark this answers is not in the document]
Comment (cathaysia) [reply lua/aerial/backends/asciidoc.lua]: I add codes check title inside block and test it manually, but symbol/asciidoc_backend.json maybe need update.

[the remark this answers is not in the document]
Comment (cathaysia) [reply lua/aerial/backends/asciidoc.lua]: I updated this json file. :)

**Comment by stevearc** [comment]
Thanks for the PR!

**Pull request merged**